external/CMakeLists.txt

```diff
@@ -19,14 +19,22 @@
 # http://numenta.org/licenses/
 # -----------------------------------------------------------------------------
 
-cmake_minimum_required(VERSION 3.3)
+cmake_minimum_required(VERSION 3.7)
 project(nupic_core CXX)
 
 set(CMAKE_VERBOSE_MAKEFILE OFF)
 include(GNUInstallDirs)
 
+message(STATUS "----- external -----")
+message(STATUS "REPOSITORY_DIR = ${REPOSITORY_DIR}")
+message(STATUS "EP_BASE    = ${EP_BASE}")
 set_directory_properties(PROPERTIES EP_BASE "${EP_BASE}")
 
+# Determine common toolchain settings, compiler and link flags
+set(INTERNAL_CPP_STANDARD C++11)
+include("${REPOSITORY_DIR}/CommonCompilerConfig.cmake")
```

Comment:
for external? I'd build external with default flags, and only use compiler confings for our sources?

Reply:
I don't like calling CommonCompilerConfig.cmake twice (once here and once in src) but I wanted to compile the external stuff in C++11 and eventually src in C++17.  So the compile flags had to be setup differently.  Using default flags sounds good but default is different for different platforms. That is basically what CommonCompilerConfig figures out.

After we get rid of SWIG and APR we might be able to trim down CommonCompilerConfig to just the basics.